newm-server/src/main/kotlin/io/newm/server/aws/AwsSecretsManagerConfigLoader.kt

```diff
@@ -0,0 +1,66 @@
+package io.newm.server.aws
+
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerAsyncClientBuilder
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest
+import com.typesafe.config.ConfigFactory
+import io.ktor.server.config.ApplicationConfig
+import io.ktor.server.config.ConfigLoader
+import io.ktor.server.config.HoconApplicationConfig
+import kotlinx.serialization.json.Json
+import java.io.File
+
+class AwsSecretsManagerConfigLoader : ConfigLoader {
+    private val secretsManager = AWSSecretsManagerAsyncClientBuilder.standard()
```

Comment:
Would love to inject these - but this code runs before the dependency injector is initialized.  Is there a work around?

Reply:
Not sure if it helps or not, but if you use `by inject()` instead of constructor injection, it gets lazily initialized and doesn't actually inject until it's used.  Maybe you can re-arrange the modules so this gets created by DI first or maybe break it out into its own module that will be configured before this code is initialized.

Reply:
Yes, that should work.